cpp/dolfinx/fem/dofmapbuilder.cpp

```diff
@@ -231,20 +240,8 @@ build_basic_dofmap(
   const int num_cells = topology.connectivity(D, 0)->num_nodes();
   const std::vector<int>& group_offsets = topology.entity_group_offsets(D);
 
-  std::vector<std::int32_t> cell_ptr;
-  cell_ptr.reserve(num_cells + 1);
-  cell_ptr.push_back(0);
-
   const std::size_t nelem = element_dof_layouts.size();
-  // Go through elements twice: regular cells followed by ghost cells.
-  for (std::size_t i = 0; i < 2 * nelem; ++i)
-  {
-    // Number of dofs per cell for this element layout
-    const int local_dim = element_dof_layouts[i % nelem].num_dofs();
-    for (int j = group_offsets[i]; j < group_offsets[i + 1]; ++j)
-      cell_ptr.push_back(cell_ptr.back() + local_dim);
-  }
-  std::vector<std::int32_t> dofs(cell_ptr.back());
+  std::vector<std::int32_t> dofs(num_cells * element_dof_layouts[0].num_dofs());
```

Comment:
Is this correct?

Above (Line 177-187) we verify that the number of dofs in each layout is the same for d=0,..<tdim, but the number of dofs in the whole layout 0 can be different from the one in layout one, as the number of dofs owned by cell can differ.

Reply:
None of the code for multiple layouts works yet.